app/javascript/components/student/editor/ExercismMonacoEditor.tsx

```diff
@@ -44,18 +44,17 @@ export function ExercismMonacoEditor({
   value: string | null | undefined
   theme: string
 }) {
-  const handleEditorDidMount = (
-    editor: monacoEditor.editor.IStandaloneCodeEditor
-  ) => {
-    editor.addAction({
-      id: 'runTests',
-      label: 'Run tests',
-      keybindings: [monacoEditor.KeyCode.F2],
-      run: onRunTests,
-    })
+  const startLanguageClient = useCallback(() => {
+    const languageServerUrl = document.querySelector<HTMLMetaElement>(
+      'meta[name=language-server-url]'
+    )?.content
 
-    MonacoServices.install(editor)
-    const url = normalizeUrl(`${process.env.LANGUAGE_SERVER_HOST}/${language}/${uuidv4()}`)
+    if (!languageServerUrl) {
+      console.error('Language server URL not found')
+      return
+    }
+
+    const url = normalizeUrl(`${languageServerUrl}/${language}/${uuidv4()}`)
```

Comment:
I would probably move the entire "find and construct the url (this line, all the way up to the start of the callback)" into a `useMemo` instead, and make this thing a `useEffect` with the `url` as dependency. This also allows you to remove the `window` effect.